hibernate-reactive-core/src/main/java/org/hibernate/reactive/persister/collection/impl/ReactiveBasicCollectionPersister.java

```diff
@@ -0,0 +1,303 @@
+				);
+			}
+
+			List<Object> params = new ArrayList<>();
+			params.add( id );
+			String sql = getSQLDeleteString();
+			return reactiveConnection.update( sql, params.toArray( new Object[0] ) );
+		}
+		return CompletionStages.completedFuture( -1 );
+	}
+
+	@Override
+	public CompletionStage<Void> reactiveDeleteRows(
+			PersistentCollection collection,
+			Serializable id,
+			SharedSessionContractImplementor session) throws HibernateException {
+
+		if ( isInverse ) {
+			return CompletionStages.voidFuture();
+		}
+
+		if ( !isRowDeleteEnabled() ) {
+			return CompletionStages.voidFuture();
+		}
+
+		CompletionStage<Integer> loop = CompletionStages.zeroFuture();
+		ReactiveConnection reactiveConnection = getReactiveConnection( session );
+
+		boolean deleteByIndex = hasIndex && !indexContainsFormula;
+
+		Iterator deletes = collection.getDeletes( this, !deleteByIndex );
+
+		int offset = 1;
+		int i = 0;
+
+		while ( deletes.hasNext() ) {
+			final Object entry = deletes.next();
+			if ( collection.entryExists( entry, i ) ) {
+				Object[] paramValues = PreparedStatementAdaptor.bind(
+						st -> {
+							int loc = offset;
+							if ( hasIdentifier ) {
+								writeIdentifier( st, entry, loc, session );
+							}
+							else {
+								loc = writeKey( st, id, loc, session );
+								if ( deleteByIndex ) {
+									writeIndexToWhere( st, entry, loc, session );
+								}
+								else {
+									writeElementToWhere( st, entry, loc, session );
+								}
+							}
+						}
+				);
+				loop = loop.thenCompose( total -> reactiveConnection
+						.update( getSQLDeleteRowString(), paramValues )
+						.thenApply( deleteCount -> total + deleteCount ) );
+			}
+		}
+
+		return loop.thenAccept( total -> LOG.debugf( "Done deleting rows: %s deleted", total ) );
+	}
+
+	@Override
+	public CompletionStage<Void> reactiveUpdateRows(
+			PersistentCollection collection,
+			Serializable id,
+			SharedSessionContractImplementor session) throws HibernateException {
+
+		// Currently there is no "updateRows()" method in AbstractCollectionPersister
+		// Not sure if this method, reactiveUpdateRows() is needed.
+
+		// TODO:  @Barry
+		// Can each element in the collection be mapped unequivocally to a single row in the database?
+		// Generally bags and sets are the only collections that cannot be.
+		// So Maps, for instance will return isRowUpdatePossible() == TRUE
+		// and require doUpdateRows() to be replaced by a reactive
+
+		if ( !isInverse && collection.isRowUpdatePossible() ) {
+
+			LOG.debugf( "Updating rows of collection: %s#%s", getNavigableRole().getFullPath(), id );
+
+			// update all the modified entries
+			// NOTE this method call uses a JDBC connection and will fail for Map ElementCollection type
+			// May also have to override getSQLDeleteString() to correctly parameterize to handle the SQL for the PG use case
+			// SEE getSQLInsertRowString() in this class for an example
+			int count = doUpdateRows( id, collection, session );
```

Comment:
This is not the correct implementation anyway so it's probably going to fail if somebody reaches this code. I think we should throw an `UnsupportedOperationException` instead like we do in reactiveDeleteRows.